python-test/features/steps/functions.py

```diff
@@ -0,0 +1,123 @@
+import requests
+import json
+from pygments import highlight, lexers, formatters
+import subprocess
+import shlex
+import re
+import time
+import docker
+
+base_orb_url = "https://beta.orb.live"
+email = 'tester@email.com'
+password = '12345678'
+
+
+def generate_token(email, password, base_orb_url = base_orb_url): 
+    headers={'Content-type':'application/json',  'Accept':'*/*'}
+    token_request = requests.post(base_orb_url+'/api/v1/tokens', json= {'email': email, 'password': password}, headers=headers)
+    if token_request.status_code >= 200 and token_request.status_code <300:
+        return (((token_request.text).split(":"))[1]).split('"')[1]
+    else:
+        return token_request.status_code
+
+
+def check_existing_agents(token, id= "" , base_orb_url = base_orb_url):
+    agentDict = dict()
+    check_agents = requests.get(base_orb_url+'/api/v1/agents'+id, headers={'Authorization': token})
+    agents_as_json = json.loads(check_agents.text)
+    formatted_json = json.dumps(agents_as_json, sort_keys=True, indent=4)
+    colorful_json = highlight(formatted_json, lexers.JsonLexer(), formatters.TerminalFormatter())
+    # print(colorful_json)
+    if id == "":
+        for i in range(len(agents_as_json['agents'])):
+            agentDict[agents_as_json['agents'][i]['name']] = {'id': agents_as_json['agents'][i]['id'], 'state': agents_as_json['agents'][i]['state']}
+    else:
+            agentDict[agents_as_json['name']] = {'id': agents_as_json['id'], 'state': agents_as_json['state']}
+    return agentDict
+
+
+def delete_agent(list_of_agents, token, base_orb_url = base_orb_url):
+    agent_name = list_of_agents.keys()
+    for agent in agent_name:
+        delete_agent = requests.delete(base_orb_url+'/api/v1/agents/'+list_of_agents[agent]['id'], headers={'Authorization': token})
+        assert delete_agent.status_code == 204
+
+def new_agent(token, name, tagKey, tagValue, base_orb_url = base_orb_url):
+    new_agent = requests.post(base_orb_url+'/api/v1/agents', json= {"name": name,"orb_tags": {tagKey: tagValue}, "validate_only": False}, headers={'Content-type':'application/json',  'Accept':'*/*', 'Authorization': token})
+    return new_agent.status_code, json.loads(new_agent.text)
+
+def send_terminal_commands(command, separator=None, cwd_run=None):
+    args = shlex.split(command)
+    docker_command_execute = subprocess.Popen(args, stdout=subprocess.PIPE, cwd=cwd_run)
```

Comment:
Is the long term strategy to use a terminal subprocess instead of interacting with docker directly?

Reply:
The way I would prefer to do it would be direct calls to docker with https://docker-py.readthedocs.io/en/stable/